segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -323,20 +327,23 @@ public void updateSegmentAttribute(UpdateSegmentAttribute updateSegmentAttribute
         long trace = LoggerHelpers.traceEnter(log, operation, updateSegmentAttribute);
         val update = new AttributeUpdate(attributeId, AttributeUpdateType.ReplaceIfEquals, newValue, expectedValue);
         segmentStore.updateAttributes(segmentName, Collections.singletonList(update), TIMEOUT)
-                .whenComplete((v, e) -> {
-                    LoggerHelpers.traceLeave(log, operation, trace, e);
-                    if (e == null) {
-                        connection.send(new SegmentAttributeUpdated(requestId, true));
-                    } else {
-                        if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
-                            log.debug("Updating segment attribute {} failed due to: {}", update, e.getMessage());
-                            connection.send(new SegmentAttributeUpdated(requestId, false));
+                    .whenComplete((v, e) -> {
+                        LoggerHelpers.traceLeave(log, operation, trace, e);
+                        final Consumer<Throwable> failureHandler = t -> {
+                            log.error(requestId, "Error (Segment = '{}', Operation = '{}')", segmentName, "handling result of " + operation, t);
+                            connection.close();
+                        };
+                        if (e == null) {
+                            invokeSafely(connection::send, new SegmentAttributeUpdated(requestId, true), failureHandler);
                         } else {
-                            handleException(requestId, segmentName, operation, e);
+                            if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
+                                log.debug("Updating segment attribute {} failed due to: {}", update, e.getMessage());
+                                invokeSafely(connection::send, new SegmentAttributeUpdated(requestId, false), failureHandler);
+                            } else {
+                                handleException(requestId, segmentName, operation, e);
+                            }
                         }
-                    }
-                })
-                .exceptionally(e -> handleException(requestId, segmentName, operation, e));
```

Comment:
`exceptionally` statement has been removed since it will not be invoked if there is an exception thrown during the execution of `whenComplete` method. Also, an exception thrown by ` segmentStore.updateAttributes` would cause both the `whenComplete` and `exceptionally` blocks to be executed.